nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/main/java/org/apache/nifi/processors/hadoop/PutHDFS.java

```diff
@@ -160,6 +170,14 @@ public class PutHDFS extends AbstractHadoopProcessor {
             .allowableValues(WRITE_AND_RENAME_AV, SIMPLE_WRITE_AV)
             .build();
 
+    public static final PropertyDescriptor APPEND_MODE = new PropertyDescriptor.Builder()
+            .name("Append Mode")
+            .description("Defines the append strategy to use when the Conflict Resolution Strategy is set to 'append'.")
+            .allowableValues(DEFAULT_APPEND_MODE, AVRO_APPEND_MODE)
+            .defaultValue(DEFAULT_APPEND_MODE)
+            .required(true)
+            .build();
```

Comment:
`Append Mode` property should depend on `Conflict Resolution Strategy` and should only be shown if the strategy is `append`.
I also suggest placing the new property just below `Conflict Resolution Strategy`.

Reply:
Fixed